File: Subastas-app/subastas-api/src/main/java/com/subastas/subastas_api/model/Moneda.java
package com.subastas.subastas_api.model;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum Moneda {
    ARS,
    PESOS,
    DOLARES;

    @JsonCreator
    public static Moneda fromValue(String value) {
        if (value == null) {
            return null;
        }

        return switch (value.trim().toUpperCase()) {
            case "ARS", "PESOS" -> ARS;
            case "USD", "DOLARES" -> DOLARES;
            default -> Moneda.valueOf(value.trim().toUpperCase());
        };
    }

    public Moneda normalizada() {
        if (this == PESOS) {
            return ARS;
        }

        return this;
    }

    public boolean esMismaMoneda(Moneda otra) {
        return otra != null && this.normalizada() == otra.normalizada();
    }
}

File: Subastas-app/subastas-client/src/utils/moneda.ts
export type BackendMoneda = "PESOS" | "DOLARES";
export type FrontendMoneda = BackendMoneda | "ARS" | "USD" | string;

export function normalizeMoneda(moneda: FrontendMoneda | null | undefined) {
  if (moneda === "ARS" || moneda === "PESOS") return "PESOS";
  if (moneda === "USD" || moneda === "DOLARES") return "DOLARES";
  return moneda ?? "";
}

export function getCurrencyCode(moneda: FrontendMoneda | null | undefined) {
  return normalizeMoneda(moneda) === "DOLARES" ? "USD" : "ARS";
}

export function getMonedaLabel(moneda: FrontendMoneda | null | undefined) {
  return getCurrencyCode(moneda);
}
